Math/NumberTheory/DirichletCharacters.hs

```diff
@@ -0,0 +1,589 @@
+-- |
+-- Module:      Math.NumberTheory.DirichletCharacters
+-- Copyright:   (c) 2018 Bhavik Mehta
+-- Licence:     MIT
+-- Maintainer:  Bhavik Mehta <bhavikmehta8@gmail.com>
+-- Stability:   Provisional
+-- Portability: Non-portable (GHC extensions)
+--
+-- Implementation and enumeration of Dirichlet characters.
+--
+
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ViewPatterns #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE KindSignatures #-}
+{-# LANGUAGE GADTs #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE CPP #-}
+
+module Math.NumberTheory.DirichletCharacters
+  (
+  -- * Roots of unity
+    RootOfUnity
+  -- ** Conversions
+  , toRootOfUnity
+  , fromRootOfUnity
+  , toComplex
+  -- * An absorbing semigroup
+  , OrZero, pattern Zero, pattern NonZero
+  , asNumber
+  -- * Dirichlet characters
+  , DirichletCharacter
+  -- ** Construction
+  , indexToChar
+  , indicesToChars
+  , characterNumber
+  , allChars
+  -- ** Evaluation
+  , evaluate
+  , generalEval
+  , toFunction
+  , allEval
```

Comment:
Let us use the same stem: either `eval`, `generalEval`, `allEval` or `evaluate`, `generalEvaluate`, `allEvaluate`. I would also put the verb first: e. g., `evalAll` and `evalGeneral`.

Reply:
Adjusted, thanks.